src/main/java/com/nirima/jenkins/plugins/docker/DockerCloud.java

```diff
@@ -731,7 +738,9 @@ protected Object readResolve() {
         }
 
         if (name == null || name.isBlank()) {
-            LOGGER.warn("Docker cloud requires a non-blank name after Jenkins 2.402");
+            String newName = "docker-cloud-" + Integer.toHexString(hashCode());
```

Comment:
`hashCode()` is not collision-free so could cause duplicate names.

Reply:
I confirmed that collisions happen when the hashCode values of two clouds are the same and the cloud name of both clouds is blank.  Since the hashCode includes the DockerAPI object and the list of templates, I think it will be very rare that a Jenkins controller has more than one docker cloud definition that have the same values for DockerAPI and template list and use blank names for the cloud name.

I think it is better to risk a collision and prefer a stable transformation from blank name to generated name.  If I remove the risk of collision, it seems like I am accepting that the transformation from blank name to generated name will choose a unique name each time it is run.

Does that seem like an acceptable risk to you or should I accept that the transformation will be different each time it is applied to a docker cloud that has a blank name?

Reply:
Worse thing that will happen is you will see the same entry twice in the cloud list, causing a weird UX. Clicking on one of them will show you the first entry appearing in the underlying persisted list.
I think it can still be renamed to a different name by the user to clear the ambiguity.
However until the rename happens, the second cloud with the same name won't be editable.

Reply:
When I did the rename in my test of editing the first in the list, it showed the new name for both in the web page.  However, I didn't spend a lot of time exploring beyond that.

I think this is ready to merge.  Interactive testing is complete.  The pull request description includes details of that testing.  Anything else that you feel should be changed before it is merged and released?